bson/bsonrw/extjson_parser.go

```diff
@@ -118,16 +118,16 @@ func (ejp *extJSONParser) peekType() (bsontype.Type, error) {
 			switch t {
 			case bsontype.JavaScript:
 				// just saw $code, need to check for $scope at same level
-				_, err := ejp.readValue(bsontype.JavaScript)
-
-				if err != nil {
+				_, tempErr := ejp.readValue(bsontype.JavaScript)
```

Comment:
why did you create a new variable instead of reassigning the `err` variable from the outer scope like the other places in this function?

Reply:
My understanding was that you'd shadowed intentionally so the error from `readValue` wouldn't be returned from this function. I changed it in my latest commit to reassign to `err` instead.

Reply:
Oh, I don't believe it was intentional when I originally wrote this. I think the error from `readValue` is what should be returned. In fact, I'm not sure how the returned error was non-nil when this was shadowed/when you had a temporary variable. The change you made is what we should be doing.

Reply:
The returned error was not non-nil when we were shadowing. This would return `nil` but we'd get an error somewhere later on because we'd try to do casting/validation and something would be wrong.